Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionCocoa.mm

```diff
@@ -122,44 +122,39 @@
 
 static const size_t maximumNumberOfShortcutCommands = 4;
 
-WebExtension::WebExtension(NSBundle *appExtensionBundle, NSError **outError)
+WebExtension::WebExtension(NSBundle *appExtensionBundle, NSURL *resourceBaseURL, NSError **outError)
     : m_bundle(appExtensionBundle)
-    , m_resourceBaseURL(appExtensionBundle.resourceURL.URLByStandardizingPath.absoluteURL)
+    , m_resourceBaseURL(resourceBaseURL)
     , m_manifestJSON(JSON::Value::null())
 {
-    ASSERT(appExtensionBundle);
+    RELEASE_ASSERT(m_bundle || m_resourceBaseURL);
 
-    if (outError)
-        *outError = nil;
+    auto *bundleResourceURL = m_bundle.get().resourceURL.URLByStandardizingPath.absoluteURL;
+    if (!m_resourceBaseURL && bundleResourceURL)
+        m_resourceBaseURL = bundleResourceURL;
 
-    if (!manifestParsedSuccessfully()) {
-        ASSERT(m_errors.get().count);
-        *outError = m_errors.get().lastObject;
-    }
-}
+#if PLATFORM(MAC)
+    m_shouldValidateResourceData = m_bundle ? [m_resourceBaseURL isEqual:bundleResourceURL] : false;
```

Comment:
Nit: This could just be:

`m_shouldValidateResourceData = m_bundle && [m_resourceBaseURL isEqual:bundleResourceURL]`